src/ServiceControl/Monitoring/EndpointInstanceId.cs

```diff
@@ -0,0 +1,47 @@
+namespace ServiceControl.Monitoring
+{
+    using System;
+    using ServiceControl.Infrastructure;
+
+    public class EndpointInstanceId : IEquatable<EndpointInstanceId>
+    {
+        public readonly string LogicalName;
+        public readonly string HostName;
+        public readonly Guid HostGuid;
+
+        public EndpointInstanceId(string logicalName, string hostName, Guid hostGuid)
+        {
+            LogicalName = logicalName;
+            HostName = hostName;
+            HostGuid = hostGuid;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != this.GetType()) return false;
+            return Equals((EndpointInstanceId)obj);
```

Comment:
@WilliamBZA I didn't write this implementation. I just used the R# generated one. This is much simpler. I'll update it